service_manager_platform_interface/lib/src/implementations/method_channel_service_manager.dart

```diff
@@ -18,4 +19,9 @@ class MethodChannelServiceManager extends ServiceManagerPlatform {
     return wasBluetoothEnabled;
   }
 
+  @override
+  Stream<dynamic> get state {
+    return eventChannel.receiveBroadcastStream();
+  }
```

Comment:
You shouldn't return `Stream<dynamic>` here but e.g. `Stream<BluetoothState>`
If you receive integers from the channel you can cast it like this:
```suggestion
   eventChannel
        .receiveBroadcastStream()
        .cast<List<int>>()
        .map((data) {
      switch (data) {
        case 10:
          return BluetoothState.off;
        case 12:
          return BluetoothState.on;
        default:
          return BluetoothState.unknown;
      }
    });
```